mlflow/projects/utils.py

```diff
@@ -0,0 +1,127 @@
+import os
+
+from mlflow.exceptions import ExecutionException
+
+MLPROJECT_PARAMETER_TYPES = ('string', 'float', 'path', 'uri')
+BAD_MLPROJECT_MESSAGE = "Invalid MLproject file: {}"
+
+
+def validate_conda_env_path(path):
+    if not os.path.exists(path):
+        bad_conda_path_message = "conda environment file {} not found".format(path)
+        raise ExecutionException(
+            BAD_MLPROJECT_MESSAGE.format(bad_conda_path_message)
+        )
+
+
+def validate_project_yaml(project_yaml):
+    """validate the yaml in an MLproject file"""
+    project_allowed_entries = ('name', 'conda_env', 'docker_env', 'entry_points')
+
+    project_entries = project_yaml.keys()
+
+    # make sure there are no extraneous entries
+    _validate_entries_are_allowed('project-level', project_entries, project_allowed_entries)
```

Comment:
@drewmcdonald @smurching I'm not sure I see the value in restricting the set of allowed MLproject keys. Why not allow users to augment their MLproject YAML with additional information? I agree that we should validate that *required* entries are present up front.

Reply:
@drewmcdonald IMO an INFO message along the lines of `_logger.info("Found non-standard top-level MLproject fields %s. Known fields are %s" % (extraneous_entries, project_entries))` makes sense, we can remove it if it proves too annoying for the few users who might add custom fields to their MLproject files